Source/Core/Gen3/Searchers/StaticSearcher3.cpp

```diff
@@ -32,28 +32,28 @@ StaticSearcher3::StaticSearcher3(Method method, const Profile3 &profile, const S
 {
 }
 
-void StaticSearcher3::startSearch(const std::array<u8, 6> &min, const std::array<u8, 6> &max, const StaticTemplate *staticTemplate)
+void StaticSearcher3::startSearch(const std::array<u8, 6> &min, const std::array<u8, 6> &max, const StaticTemplate *staticTemplate, const bool buggedRoamer)
 {
     searching = true;
 
     for (u8 hp = min[0]; hp <= max[0]; hp++)
     {
-        for (u8 atk = min[1]; atk <= max[1]; atk++)
+        for (u8 atk = buggedRoamer ? min[1] & 7 : min[1]; atk <= (buggedRoamer ? max[1] & 7 : max[1]); atk++)
         {
-            for (u8 def = min[2]; def <= max[2]; def++)
+            for (u8 def = buggedRoamer ? 0 : min[2]; def <= (buggedRoamer ? 31 : max[2]); def++)
             {
-                for (u8 spa = min[3]; spa <= max[3]; spa++)
+                for (u8 spa = buggedRoamer ? 0 : min[3]; spa <= (buggedRoamer ? 31 : max[3]); spa++)
                 {
-                    for (u8 spd = min[4]; spd <= max[4]; spd++)
+                    for (u8 spd = buggedRoamer ? 0 : min[4]; spd <= (buggedRoamer ? 31 : max[4]); spd++)
                     {
-                        for (u8 spe = min[5]; spe <= max[5]; spe++)
+                        for (u8 spe = buggedRoamer ? 0 : min[5]; spe <= (buggedRoamer ? 31 : max[5]); spe++)
```

Comment:
I don't like this approach. It limits how many origin seeds we can search through. See how I handled similar situation for the e-readers in colo https://github.com/Admiral-Fish/PokeFinder/blob/master/Source/Core/Gen3/Searchers/GameCubeSearcher.cpp#L300-L307